src/access/acp.test.ts

```diff
@@ -3653,3 +3686,250 @@ describe("internal_getActorAccessAll", () => {
+              read: true,
+            },
+          },
+        },
+        memberPolicies: {},
+        acrPolicies: {},
+        memberAcrPolicies: {},
+      }
+    );
+
+    const access = internal_getGroupAccessAll(resourceWithAcr);
+
+    expect(access).toStrictEqual({});
+  });
+});
+
+describe("getAgentAccessAll", () => {
+  const agentAUrl = "https://some.pod/profiles#agentA";
+  const agentBUrl = "https://some.pod/profiles#agentB";
+
+  it("returns access set for any Agent referenced in the ACR", () => {
+    const resourceWithAcr = mockResourceWithAcr(
+      "https://some.pod/resource",
+      "https://some.pod/resource?ext=acr",
+      {
+        policies: {
+          "https://some.pod/resource?ext=acr#policy": {
+            allOf: {
+              "https://some.pod/resource?ext=acr#applicable-allOf-rule": {
+                [acp.agent]: [agentAUrl, agentBUrl],
+              },
+            },
+            allow: {
+              read: true,
+            },
+          },
+        },
+        memberPolicies: {},
+        acrPolicies: {},
+        memberAcrPolicies: {},
+      }
+    );
+
+    const access = internal_getAgentAccessAll(resourceWithAcr);
+
+    expect(access).toStrictEqual({
+      [agentAUrl]: {
+        read: true,
+      },
+      [agentBUrl]: {
+        read: true,
+      },
+    });
+  });
+
+  it("does not return access set for a group", () => {
+    const resourceWithAcr = mockResourceWithAcr(
+      "https://some.pod/resource",
+      "https://some.pod/resource?ext=acr",
+      {
+        policies: {
+          "https://some.pod/resource?ext=acr#policy": {
+            allOf: {
+              "https://some.pod/resource?ext=acr#applicable-allOf-rule": {
+                [acp.group]: ["https://some.pod/group#some-group"],
+              },
+            },
+            allow: {
+              read: true,
+            },
+          },
+        },
+        memberPolicies: {},
+        acrPolicies: {},
+        memberAcrPolicies: {},
+      }
+    );
+
+    const access = internal_getAgentAccessAll(resourceWithAcr);
+
+    expect(access).toStrictEqual({});
+  });
+
+  it("does not include access set for the public agent", () => {
+    const resourceWithAcr = mockResourceWithAcr(
+      "https://some.pod/resource",
+      "https://some.pod/resource?ext=acr",
+      {
+        policies: {
+          "https://some.pod/resource?ext=acr#policy": {
+            allOf: {
+              "https://some.pod/resource?ext=acr#applicable-allOf-rule": {
+                [acp.agent]: [acp.PublicAgent],
+              },
+            },
+            allow: {
+              read: true,
+            },
+          },
+        },
+        memberPolicies: {},
+        acrPolicies: {},
+        memberAcrPolicies: {},
+      }
+    );
+
+    const access = internal_getAgentAccessAll(resourceWithAcr);
+
+    expect(access).toStrictEqual({});
+  });
+
+  it("does not return access set for any authenticated Agent", () => {
```

Comment:
![Ditto](https://user-images.githubusercontent.com/4251/105966178-98c61500-6084-11eb-9b85-a80e3d50d5f6.png)

for the Creator.

Reply:
:) - but surely you mean **_the_** creator: https://www.dw.com/en/spaghetti-monster-goes-to-germanys-constitutional-court/a-40359916

Reply:
It took me a trip to Poképedia to get the joke, but I love it. Do pokemons have Dutch names as well ?

Reply:
66db8588f502db2fb8806c86e1bfac562a83382e